api/proxy/config/v2/eigendaflags/cli.go

```diff
@@ -190,14 +191,13 @@ loaded into memory for KZG commitments. Example units: '15MiB', '4Kib'.`,
 		},
 		&cli.StringFlag{
 			Name: NetworkFlagName,
-			Usage: fmt.Sprintf(`The EigenDA network that is being used. This is an optional flag, to configure
-default values for %s, %s, and %s. If all of these fields are explicitly configured, the
-network flag may be omitted. If some or all of these fields are configured, and the network
-is also configured, then the explicitly defined field values will take precedence. Permitted
-EigenDANetwork values include %s, %s, %s, & %s.`,
-				DisperserFlagName,
-				ServiceManagerAddrFlagName,
-				BLSOperatorStateRetrieverFlagName,
+			Usage: fmt.Sprintf(`The EigenDA network that is being used. This is an optional flag, 
+to configure default values for different EigenDA contracts and disperser URL. 
+See https://github.com/Layr-Labs/eigenda/blob/master/api/proxy/common/eigenda_network.go
+for the exact values getting set by this flag. All of those values can also be manually
+set via their respective flags, and take precedence over the default values set by the network flag.
+If all of those other flags are manually configured, the network flag may be omitted. 
+Permitted EigenDANetwork values include %s, %s, %s, & %s.`,
 				common.MainnetEigenDANetwork,
 				common.HoleskyTestnetEigenDANetwork,
 				common.HoleskyPreprodEigenDANetwork,
```

Comment:
Here's a thought:

right now, the `eigenda_network` is a bit hard to maintain. If you want to add a network type, you have to define the name constant, and then make sure you add the correct values to each switch statement in that file.

It would be a lot easier if we just defined an `EigenDANetwork` struct type, which contains all the necessary fields. Then all values for a given network would be defined together.

Additionally, if we maintained a list of all existing `EigenDANetwork` types in `eigenda_network.go`, we could use it for logs like this (join all the names with `, `), and then adding a new network type wouldn't entail modifying multiple files, it would all be contained in `eigenda_network.go`

This isn't necessarily something that needs to be done right now, or at all.

Reply:
I 200% agree with you. Code that isn't meant to be dynamically dispatched should not be written with that flexibility, because it loses the static typing guarantees.
I do want to change this, but will do in a followup PR: https://linear.app/eigenlabs/issue/DAINT-662/make-eigenda-networks-statically-typed